### PR TITLE
Fix reading serial slave response RTU

### LINKED ISSUE #76: Serial slave not reading response from other slave
## Expected Behavior

The slave should detect that the master request is not destined to the slave. After this the slave should expect there to come a response on the serial line and read it. Otherwise should timeout and expect another request.

## Actual Behavior

The slave detects that the message it not for it, and correctly ignores it. But the slave does not expect the response and thinks that this is supposed to be another request. This results in the slave not correctly reading the next request meant for it.

## Steps to Reproduce the Problem

  1. Connect one master to two slaves on a serial bus.
  2. Send read requests to one of the slaves.
  3. Look at debug printouts on the other master.

## Specifications

  - Version: 2.5.1-SNAPSHOT

### PR DESCRIPTION
**Pull request by NogginBops**
This PR makes it so that the slave implementation expects an answer for messages not meant for it.

This is more of a proof of concept more than a full pr. The same thing should be applied to Modbus ASCII and thought should be put into what to do if there never comes a response. (Maybe a timeout)

This also clarifies some debug printouts that were wrong too. (and adds some for the new feature 🙂)

Fixes #76

### CODE REVIEW

**Comment by NogginBops** [comment]
This fix does not work atm as if the master resends it's request as the slave is expecting a response the slave doesn't work.

**Comment by NogginBops** [comment]
There needs to be a good way o either detect if a message is a request or a response, or there needs to be a good way to ignore all messages that are not meant for the slave (Using the 3.5T break as the end of the message).

**Comment by steveohara** [comment]
I can't accept this PR I'm afraid;

- It must be built for JDK 1.6 (it contains some JDK 8 code)
- You need to remove the extraneous JAR files from the root. The build is a Maven project and any required libraries should be maintained through that mechanism
- We are not distributing modpoll - it is used purely for testing so should not be included in the main source tree

If you fix these issues, I'll take another look at merging it into development and then production.
Thanks

File: src/main/java/com/ghgande/j2mod/modbus/io/ModbusSerialTransport.java [hunk 1]
```diff
@@ -591,19 +601,68 @@ void waitBetweenFrames(int transDelayMS, long lastTransactionTimestamp) {
...
+        }
+    }
+    
+    /**
+     * The maximum delay between characters in microseconds
+     * @return
+     */
+    long getMaxCharDelay() {
+        if (commPort.getBaudRate() > 19200) {
+            return 1750;
```
Comment (NogginBops) [inline src/main/java/com/ghgande/j2mod/modbus/io/ModbusSerialTransport.java]: This should be 750 from the modbus specification but that did not seem to work in practice, probably OS delays and other factors. Should this delay be configurable... ? Seems kind of overkill

**Comment by steveohara** [comment]
Your branch still has 1.8 classes in it e.g. `BooleanSupplier`

**Pull request closed without merging**